source/rtos_none/SysTick_Handler.c

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file    SysTick_Handler.c
+  * @brief  No RTOS port only needs a systick handler
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+ #include "rtos_api.h"
+
+/* NVIC registers */
+#define NVIC_ST_CTRL    (*((volatile U32 *)0xE000E010))
+#define NVIC_ST_RELOAD  (*((volatile U32 *)0xE000E014))
+#define NVIC_ST_CURRENT (*((volatile U32 *)0xE000E018))
+
+// </h>
+// <h>SysTick Timer Configuration
+// =============================
+//   <o>Timer clock value [Hz] <1-1000000000>
+//   <i> Set the timer clock value for selected timer.
+#ifndef OS_CLOCK
+#error "OS_CLOCK should be defined by HIC configuration"
+#endif
+
+//   <o>Timer tick value [us] <1-1000000>
+//   <i> Set the timer tick value for selected timer.
+#ifndef OS_TICK
+#error "OS_TICK should be defined by RTOS configuration"
+#endif
+
+#define OS_TRV          ((U32)(((double)OS_CLOCK*(double)OS_TICK)/1E6)-1)
+
+
+static volatile rtosTick_t tick_counter = 0;
+static volatile rtosEvent_t event_flags=0;
+static volatile rtosTick_t wait_counter = 0;
+
+static sysTickCb_t sysTickCb = NULL;
+static rtosTick_t tickFreq = 1;
+
+void sysTickInit(void)
+{
+  NVIC_ST_RELOAD  = OS_TRV;
+  NVIC_ST_CURRENT = 0;
+  NVIC_ST_CTRL    = 0x0007;
```

Comment:
Use the SysTick related register definitions and functions in the CMSIS-Core headers instead of writing your own. See [core_cm4.h:647](https://github.com/ARMmbed/DAPLink/blob/master/source/cmsis-core/core_cm4.h#L647) and [core_cm4.h:1695](https://github.com/ARMmbed/DAPLink/blob/master/source/cmsis-core/core_cm4.h#L1695).